resources/qml/Sidebar.qml

```diff
@@ -214,7 +214,7 @@ Rectangle
                 anchors.left: parent.left
                 anchors.leftMargin: model.index * (settingsModeSelection.width / 2)
                 anchors.verticalCenter: parent.verticalCenter
-                width: parent.width / 2
+                width: 0.5 * parent.width - (modesListModel.get(index).showFilterButton ? toggleFilterButton.width : 0)
```

Comment:
Since you're already changing the width of the toggleFilterButton there is no need to add the extra check here, especially since it make the expression a lot more complicated. 

Also, this will remove the button width twice due to operator precedence.